packages/server/hook-require.js

```diff
@@ -0,0 +1,35 @@
+const path = require('path')
+const env = process.env.CYPRESS_INTERNAL_ENV === 'production' ? 'prod' : 'dev'
+
+process.env.PROJECT_BASE_DIR = process.env.PROJECT_BASE_DIR ?? path.join(__dirname, '..', '..')
+
+const isDev = env === 'dev'
+
+function runWithSnapshot () {
+  const { snapshotRequire } = require('@packages/v8-snapshot-require')
+  const projectBaseDir = process.env.PROJECT_BASE_DIR
+
+  snapshotRequire(projectBaseDir, {
+    diagnostics: isDev,
+    useCache: true,
+    transpileOpts: {
+      supportTS: isDev,
+      initTranspileCache: isDev
+        ? () => require('dirt-simple-file-cache').DirtSimpleFileCache.initSync(projectBaseDir, { cacheDir: path.join(projectBaseDir, 'node_modules', '.dsfc'), keepInMemoryCache: true })
+        : function () {},
```

Comment:
could we just pass undefined here instead of an empty function?

Reply:
[cypress-io/cypress@`ec34a53` (#23732)](https://github.com/cypress-io/cypress/pull/23732/commits/ec34a537a4879edcfbc0409b0f81f16a94e97ee9)